src/Tribe/Views/V2/Widgets/List_Widget.php

```diff
@@ -0,0 +1,169 @@
+			'classname'   => 'tribe-events-list-widget',
+			'description' => null
+		],
+		'control_options'      => [
+			'id_base' => 'tribe-events-list-widget'
+		],
+	];
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected $validate_arguments_map = [
+		'should_manage_url'    => 'tribe_is_truthy',
+		'no_upcoming_events'   => 'tribe_is_truthy',
+		'featured_events_only' => 'tribe_is_truthy',
+		'jsonld_enable'        => 'tribe_is_truthy',
+	];
+
+	public function get_arguments() {
+		$arguments = $this->default_arguments;
+
+		$arguments['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
+		$arguments['name'   ]         = esc_html__( 'Events List V2', 'the-events-calendar' );
+		$arguments['widget_options']['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
+
+		return $this->filter_arguments( $arguments );
+	}
+
+	/**
+	 * Alters the shortcode context with its arguments.
+	 *
+	 * @since  TBD
+	 *
+	 * @param \Tribe__Context $context Context we will use to build the view.
+	 *
+	 * @return \Tribe__Context Context after shortcodes changes.
+	 */
+	public function alter_context( Context $context, array $arguments = [] ) {
+
+		$alter_context = $this->args_to_context( $arguments, $context );
+
+		// The View will consume this information on initial state.
+		$alter_context['id']        = $this->slug;
+
+		$context = $context->alter( $alter_context );
+
+		return $context;
+	}
+
+	/**
+	 * Translates shortcode arguments to their Context argument counterpart.
+	 *
+	 * @since TBD
+	 *
+	 * @param array   $arguments The shortcode arguments to translate.
+	 * @param Context $context The request context.
+	 *
+	 * @return array The translated shortcode arguments.
+	 */
+	protected function args_to_context( array $arguments, Context $context ) {
+		$context_args = [];
+
+		$category_input = Arr::get_first_set( $arguments, [ 'cat', 'category' ], false );
+
+		if ( ! empty( $category_input ) ) {
+			$context_args['event_category'] = Arr::list_to_array( $category_input );
+		}
+
+		if ( ! empty( $arguments['date'] ) ) {
+			$context_args['event_date'] = $arguments['date'];
+		}
+
+		if ( isset( $arguments['featured'] ) ) {
+			$context_args['featured'] = tribe_is_truthy( $arguments['featured'] );
+		}
+
+		if ( null === $context->get( 'eventDisplay' ) ) {
+			if ( empty( $arguments['view'] ) ) {
+				$default_view_class                 = tribe( Views_Manager::class )->get_default_view();
+				$context_args['event_display_mode'] = tribe( Views_Manager::class )->get_view_slug_by_class( $default_view_class );
+			} else {
+				$context_args['event_display_mode'] = $arguments['view'];
+			}
+		}
+
+		return $context_args;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_html() {
+		$html_test = '<p>V2 List Widget</p>';
+
+		$context = tribe_context();
+
+		// Modifies the Context for the shortcode params.
+		$context   = $this->alter_context( $context );
+
```

Comment:
:warning: Equals sign not aligned correctly; expected 1 space but found 3 spaces

<sub>hash: 74231f777514c2bc81534e88aa2a3681</sub>